tests/test_Remote.cc

```diff
@@ -0,0 +1,110 @@
+/**
+ * \file   test_Remote.cc
+ * \author Lars Fröhlich
+ * \date   Created on January 15, 2024
+ * \brief  Test suite for the Remote class.
+ *
+ * \copyright Copyright 2024 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include <filesystem>
+#include <fstream>
+
+#include <git2.h>
+#include <gul14/catch.h>
+#include <gul14/gul.h>
+
+#include "libgit4cpp/Error.h"
+#include "libgit4cpp/GitRepository.h"
+#include "libgit4cpp/Remote.h"
+#include "libgit4cpp/wrapper_functions.h"
+#include "test_main.h"
+
+using namespace git;
+using namespace std::literals;
+using gul14::cat;
+
+TEST_CASE("Remote: Constructor", "[Remote]")
+{
+    const auto reporoot = unit_test_folder() / "Remote";
+
+    std::filesystem::remove_all(reporoot);
```

Comment:
This, and also lines 72/73:

Not sure why this code is in here.
If the tests are started the whole `unit_test_folder` get purged anyhow.
If this removing is needed, that means we reuse the directory for two tests (or at least expect that).
I would rather keep all tests' repositories in place so that one can see all artifacts (git repos) after the tests run via `cd ...; git remote -v` etc pp.
So I would NOT remove them here beforehand because that might prevent us from detecting someone who reuses the directory for another test.

Maybe the code is from before
```
tests/test_main.cc:    std::filesystem::remove_all(unit_test_folder());
```
and we have the same approach (remove individually) with `test_GitRepository.cc`. :thinking:

Reply:
Good point, removing the subdirectories is not necessary. Changed accordingly.